test/unittests/commands/xmlupload/test_transform_resource_and_values_for_upload.py

```diff
@@ -0,0 +1,425 @@
+import pytest
+import regex
+from lxml import etree
+from rdflib import RDF
+from rdflib import XSD
+from rdflib import BNode
+from rdflib import Literal
+from rdflib import Namespace
+from rdflib import URIRef
+
+from dsp_tools.commands.xmlupload.iri_resolver import IriResolver
+from dsp_tools.commands.xmlupload.models.deserialise.deserialise_value import XMLProperty
+from dsp_tools.commands.xmlupload.models.lookup_models import JSONLDContext
+from dsp_tools.commands.xmlupload.models.lookup_models import Lookups
+from dsp_tools.commands.xmlupload.models.permission import Permissions
+from dsp_tools.commands.xmlupload.models.permission import PermissionValue
+from dsp_tools.commands.xmlupload.transform_resource_and_values_for_upload import KNORA_API
+from dsp_tools.commands.xmlupload.transform_resource_and_values_for_upload import _make_one_prop_graph
+from dsp_tools.models.exceptions import BaseError
+from dsp_tools.models.exceptions import InputError
+from dsp_tools.models.exceptions import PermissionNotExistsError
+from dsp_tools.models.exceptions import UserError
+
+ONTO = Namespace("http://0.0.0.0:3333/ontology/9999/onto/v2#")
+namespaces = {"onto": ONTO, "knora-api": KNORA_API}
+
+PERMISSION_LITERAL = Literal("CR knora-admin:ProjectAdmin", datatype=XSD.string)
+RES_ONE_URI = URIRef("http://rdfh.ch/9999/res_one")
+
+
+@pytest.fixture
+def permissions_lookup() -> dict[str, Permissions]:
+    return {"open": Permissions({PermissionValue.CR: ["knora-admin:ProjectAdmin"]})}
+
+
+@pytest.fixture
+def lookups(permissions_lookup: dict[str, Permissions]) -> Lookups:
+    return Lookups(
+        project_iri="http://rdfh.ch/9999/project",
+        id_to_iri=IriResolver({"res_one": "http://rdfh.ch/9999/res_one"}),
+        permissions=permissions_lookup,
+        listnodes={"testlist:node": "http://rdfh.ch/9999/node"},
+        namespaces=namespaces,
+        jsonld_context=JSONLDContext({}),
+    )
+
+
+@pytest.fixture
+def res_info() -> tuple[BNode, str]:
+    return BNode(), "restype"
+
+
+class TestMakeOnePropGraphSuccess:
+    def test_boolean(self, lookups: Lookups, res_info: tuple[BNode, str]) -> None:
+        res_bn, res_type = res_info
+        xml_prop = etree.fromstring("""
+        <boolean-prop name=":isTrueOrFalse">
+            <boolean permissions="open">true</boolean>
+        </boolean-prop>
```

Comment:
Just a suggestion to make it more readable.

```suggestion
            <boolean-prop name=":isTrueOrFalse">
                <boolean permissions="open">true</boolean>
            </boolean-prop>
```